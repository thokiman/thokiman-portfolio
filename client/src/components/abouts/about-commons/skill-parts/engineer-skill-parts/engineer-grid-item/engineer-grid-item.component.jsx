import React from 'react';
import { DescriptionEngineerSkill } from './engineer-grid-item.styles';

export const EngineerGridItem = ({ itemsPoint }) => {
  return itemsPoint.map(({ id, description }) => {
    return (
      <DescriptionEngineerSkill className='description-engineer-skill' key={id}>
        {description}
      </DescriptionEngineerSkill>
    );
  });
};

export default EngineerGridItem;
