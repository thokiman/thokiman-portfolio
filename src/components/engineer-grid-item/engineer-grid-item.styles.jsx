import styled, { css } from "styled-components";

const EngineerSkillStyles = css`
  grid-row: span 3;
  text-align: center;
  font-kerning: auto;
  color: #e3e1e4;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const DescriptionEngineerSkill = styled.div`
  ${EngineerSkillStyles}
  line-height: 1.5em;
  width: 100%;
  font-size: 2.3vw;
  letter-spacing: 0.03em;
  @media only screen and (max-width: 800px) {
    font-size: 15px;
  }
`;

DescriptionEngineerSkill.displayName = "DescriptionEngineerSkill";
