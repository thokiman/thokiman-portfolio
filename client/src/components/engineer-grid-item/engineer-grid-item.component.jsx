import React from "react";

import { DescriptionEngineerSkill } from "./engineer-grid-item.styles";

export const EngineerGridItem = ({ itemsPoint }) => {
  return itemsPoint.map(({ id, description }) => {
    return (
      <DescriptionEngineerSkill key={id}>
        {description}
      </DescriptionEngineerSkill>
    );
  });
};

export default EngineerGridItem;
