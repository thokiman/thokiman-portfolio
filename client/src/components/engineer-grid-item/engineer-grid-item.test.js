import React from "react";
import { shallow } from "enzyme";

import { EngineerGridItem } from "./engineer-grid-item.component";

describe("EngineerGridItem component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      itemsPoint: mockArray,
    };
    wrapper = shallow(<EngineerGridItem {...mockProps} />);
  });

  it("should render EngineerGridItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
