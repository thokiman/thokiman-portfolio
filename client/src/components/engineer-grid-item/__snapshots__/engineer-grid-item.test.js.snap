// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineerGridItem component should render EngineerGridItem component 1`] = `ShallowWrapper {}`;
